modules/eslint-plugin/src/rules/store/prefix-selectors-with-select.ts

```diff
@@ -31,64 +31,180 @@ export default createRule<Options, MessageIds>({
   },
   defaultOptions: [],
   create: (context) => {
-    return {
-      'VariableDeclarator[id.name!=/^select[^a-z].+$/]:not(:has(Identifier[name="createFeature"])):matches([id.typeAnnotation.typeAnnotation.typeName.name=/^MemoizedSelector(WithProps)?$/], :has(CallExpression[callee.name=/^(create(Feature)?Selector|createSelectorFactory)$/]))'({
-        id,
-      }: TSESTree.VariableDeclarator & { id: TSESTree.Identifier }) {
-        const suggestedName = getSuggestedName(id.name);
+    function reportIfInvalid(name: string, node: TSESTree.Identifier) {
+      const isValid =
```

Comment:
I'm not as fluent in RegExp, so this comment will help me in the future.

```suggestion
      // Name starts with select and 
      // the first character after select is an uppercase ASCII letter, _, or $
      const isValid =
        name.startsWith('select') &&
        name.length > 'select'.length &&
        /^[A-Z_$]/.test(name.slice('select'.length));
```